sympy/core/function.py

```diff
@@ -1806,6 +1806,17 @@ def _eval_is_commutative(self):
         return self.expr.is_commutative
 
     def doit(self):
+        if isinstance(self.expr,Derivative):
+           if len(self.variables) == 1:
+              if sympify(sqrt(self.variables[0]**2)-self.variables[0]) == sympify(Abs(self.variables[0])-self.variables[0]):
```

Comment:
Did you want to test if the variable is real? `self.variables[0].is_real` would do that.